### PR TITLE
Return datatypes along with column names

### PR DESCRIPTION
**Pull request by KKcorps**
Currently, JDBC connector doesn't return data types along with the column names in the Result Metadata.
This causes issues in certain integrations such as Tableau.

### CODE REVIEW

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ public PinotResultSet(org.apache.pinot.client.ResultSet resultSet) {
     _closed = false;
     for (int i = 0; i < _totalColumns; i++) {
       _columns.put(_resultSet.getColumnName(i), i + 1);
+      _columnDataTypes.put(i+1, _resultSet.getColumnDataType(i));
```
Comment (fx19880617) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java:65]: nit(formatting): `i + 1`

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/Constants.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
   public static final String[] SCHEMA_COLUMNS_DTYPES = {"STRING", "STRING"};
 
   public static final String[] TABLE_COLUMNS = {"TABLE_SCHEM", "TABLE_CATALOG", "TABLE_NAME", "TABLE_TYPE", "REMARKS", "TYPE_CAT", "TYPE_SCHEM", "TYPE_NAME", "SELF_REFERENCING_COL_NAME", "REF_GENERATION"};
-  public static final String[] TABLE_COLUMNS_DTYPES = {"STRING", "STRING","STRING", "STRING","STRING", "STRING","STRING", "STRING","STRING"};
+  public static final String[] TABLE_COLUMNS_DTYPES = {"STRING", "STRING","STRING", "STRING","STRING", "STRING","STRING", "STRING","STRING","STRING"};
```
Comment (fx19880617) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/Constants.java:36]: nit: space between comma and `"STRING"`

File: pinot-clients/pinot-jdbc-client/src/test/java/org/apache/pinot/client/PinotResultSetTest.java [hunk 3]
```diff
@@ -149,6 +149,18 @@ public void testFindColumn()
...
 
+  @Test
+  public void testGetResultMetadata() throws Exception {
+    ResultSetGroup resultSetGroup = getResultSet(TEST_RESULT_SET_RESOURCE);
+    ResultSet resultSet = resultSetGroup.getResultSet(0);
+    PinotResultSet pinotResultSet = new PinotResultSet(resultSet);
+    ResultSetMetaData pinotResultSetMetadata = pinotResultSet.getMetaData();
+
+    for (int i = 0; i < resultSet.getColumnCount(); i++) {
+      Assert.assertEquals(pinotResultSetMetadata.getColumnTypeName(i+1), resultSet.getColumnDataType(i));
```
Comment (fx19880617) [inline pinot-clients/pinot-jdbc-client/src/test/java/org/apache/pinot/client/PinotResultSetTest.java:160]: nit: `i + 1`

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java (hunk 1 above)
Comment (KKcorps) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java]: done

File: pinot-clients/pinot-jdbc-client/src/test/java/org/apache/pinot/client/PinotResultSetTest.java (hunk 3 above)
Comment (KKcorps) [reply pinot-clients/pinot-jdbc-client/src/test/java/org/apache/pinot/client/PinotResultSetTest.java]: done

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/Constants.java (hunk 2 above)
Comment (KKcorps) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/Constants.java]: done

**Pull request merged**